master/internal/stream/system.go

```diff
@@ -324,28 +421,40 @@ func doPublishLoop[T stream.Msg](
 			}
 			// Broadcast all the events.
 			publisher.Broadcast(events)
-			break
 		}
 	}
-
-	return nil
 }
 
-func NewSubscriptionSet(streamer *stream.Streamer, ps PublisherSet) SubscriptionSet {
+// NewSubscriptionSet constructor for SubscriptionSet.
+func NewSubscriptionSet(
+	ctx context.Context,
+	streamer *stream.Streamer,
+	ps *PublisherSet,
+	user model.User,
+) (SubscriptionSet, error) {
+	trialPermissionFilter, err := TrialMakePermissionFilter(ctx, user)
+	if err != nil {
```

Comment:
This is going to get gross when we add more streamable entities; however, I'm not seeing not a clean way to do this (i.e. MakePermissionFilter) without moving it out of the `Subscription` constructor.

ideas:
- have `Subscription` constructor call `TrialMakePermissionFilter` instead
    - but that will result in each `Subscription` having the same `if err != nil` at a subscription level instead of a permission filter level 
- Move `TrialMakePermissionFilter` into the `SubscriptionState` constructor
    - defeats the point of enforcing that the PermissionFilter is set once on initialization.
- build a list of `*MakePermissionFilter` &  a map of permissionFilters, and then populate the SubscriptionSet using that map
  - feels like over engineering

Reply:
well, every method of SubscriptionSet has a helper function with non-idiomatic go behaviors in order to make the code nice and readable;  `.Startup()` has `startup()`, and `.SubscriptionMod()` has `subMod()`.

You could do the same thing here:

```go
func newPermFilter[T stream.Msg](
	err *error,
	permFilterFn func(context.Context, model.User)(func(T)bool, error)
	ctx context.Context,
	user model.User,
) func(T)bool {
	if *err != nil {
		return nil
	}
	out, tempErr := permFilterFn(ctx, user)
	return out
}

// NewSubscriptionSet constructor for SubscriptionSet.
func NewSubscriptionSet(
	ctx context.Context,
	streamer *stream.Streamer,
	ps *PublisherSet,
	user model.User,
) (SubscriptionSet, error) {
	var err error
	return SubscriptionSet{
		Trials: &subscriptionState[*TrialMsg, TrialSubscriptionSpec]{
			stream.NewSubscription(
				streamer,
				ps.Trials,
				newPermFilter(&err, TrialMakePermissionFilter, ctx, user),
			),
			NewTrialFilterMaker(),
			TrialCollectStartupMsgs,
			TrialCollectSubscriptionModMsgs,
		},
	}, err
}
```